### PR TITLE
Add dispensersPlaceBoatsOnIce rule and functionality

### PR DESCRIPTION
**Pull request by PrivacyPolicy**
Dispensing a boat over ice places the boat in a similar way to how boats are dispensed in water in vanilla. Useful for nether tunnels.
dispensers_place_boats_on_ice

### CODE REVIEW

File: src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import net.minecraft.util.math.BlockPointer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+
+import java.util.Map;
+
+public class PlaceBoatOnIceDispenserBehavior extends FallibleItemDispenserBehavior {
+
+    private static final Map<Item, BoatEntity.Type> BOAT_ITEM_TO_TYPE = Map.of(
+            Items.SPRUCE_BOAT, BoatEntity.Type.SPRUCE,
```
Comment (PrivacyPolicy) [inline src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java:23]: Is there a smarter way to achieve this? BoatItem.type is private, so I don't know if there's a way to obtain it directly.

Comment (Ghoulboy78) [reply src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java:23]: You could do `boatEntity.setBoatType(BoatEntity.Type.getType(item.getName().asString()));`, which also accounts for modded boat types.

Comment (PrivacyPolicy) [reply src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java:23]: I like this idea a lot, but it doesn't seem to work quite yet:
image
Seems that the `getName().asString()` just return nothing.
Looks like the only field I can get is translation key, which ends up like this:
image
Would this be preferable to the above? If so I'll commit it

Comment (PrivacyPolicy) [reply src/main/java/carpetextra/dispenser/behaviors/PlaceBoatOnIceDispenserBehavior.java]: Nevermind, I figured out a better solution! TIL mixins allow you to get private fields

**Review by altrisi — APPROVED** [review]
Very sorry for the super late response! Very good work, had to make some small changes because since Mojang changed like every tag-related line.
I'll look into making it work for the new chest boats later.

Thanks for contributing!

**Pull request merged**